File: src/main/java/net/Chidoziealways/everythingjapanese/structure/ModStructureSets.java
package net.Chidoziealways.everythingjapanese.structure;

import net.Chidoziealways.everythingjapanese.EverythingJapanese;
import net.minecraft.core.HolderGetter;
import net.minecraft.core.registries.Registries;
import net.minecraft.data.worldgen.BootstrapContext;
import net.minecraft.resources.ResourceKey;
import net.minecraft.resources.ResourceLocation;
import net.minecraft.world.level.biome.Biome;
import net.minecraft.world.level.levelgen.structure.Structure;
import net.minecraft.world.level.levelgen.structure.StructureSet;
import net.minecraft.world.level.levelgen.structure.placement.RandomSpreadStructurePlacement;
import net.minecraft.world.level.levelgen.structure.placement.RandomSpreadType;

public class ModStructureSets {
    public static ResourceKey<StructureSet> HELL_TEMPLE = register("hell_temple");

    public static void bootstrap(BootstrapContext<StructureSet> context) {
        HolderGetter<Structure> holdergetter = context.lookup(Registries.STRUCTURE);
        HolderGetter<Biome> holdergetter1 = context.lookup(Registries.BIOME);

        context.register(
                ModStructureSets.HELL_TEMPLE,
                new StructureSet(holdergetter.getOrThrow(ModStructures.HELL_TEMPLE), new RandomSpreadStructurePlacement(24, 8, RandomSpreadType.LINEAR, 20083232))
        );
    }

    private static ResourceKey<StructureSet> register(String pName) {
        return ResourceKey.create(Registries.STRUCTURE_SET, ResourceLocation.fromNamespaceAndPath(EverythingJapanese.MOD_ID, pName));
    }
}

File: src/main/kotlin/net/Chidoziealways/everythingjapanese/structure/ModStructureSets.kt
package net.Chidoziealways.everythingjapanese.structure

import net.Chidoziealways.everythingjapanese.JAPANESE_MOD_ID
import net.minecraft.core.registries.Registries
import net.minecraft.data.worldgen.BootstrapContext
import net.minecraft.resources.ResourceKey
import net.minecraft.resources.Identifier
import net.minecraft.world.level.levelgen.structure.StructureSet
import net.minecraft.world.level.levelgen.structure.placement.RandomSpreadStructurePlacement
import net.minecraft.world.level.levelgen.structure.placement.RandomSpreadType

object ModStructureSets {
    var HELL_TEMPLE: ResourceKey<StructureSet> = register("hell_temple")

    var DOJO: ResourceKey<StructureSet> = register("dojo")

    var SHOJI_HOUSE: ResourceKey<StructureSet> = register("shoji_house")

    fun bootstrap(context: BootstrapContext<StructureSet>) {
        val holdergetter = context.lookup(Registries.STRUCTURE)
        val holdergetter1 = context.lookup(Registries.BIOME)

        context.register(
            HELL_TEMPLE,
            StructureSet(
                holdergetter.getOrThrow(ModStructures.HELL_TEMPLE),
                RandomSpreadStructurePlacement(24, 8, RandomSpreadType.LINEAR, 20083232)
            )
        )

        context.register(
            DOJO,
            StructureSet(
                holdergetter.getOrThrow(ModStructures.DOJO),
                RandomSpreadStructurePlacement(24, 8, RandomSpreadType.LINEAR, 20083232)
            )
        )

        context.register(
            SHOJI_HOUSE,
            StructureSet(
                holdergetter.getOrThrow(ModStructures.SHOJI_HOUSE),
                RandomSpreadStructurePlacement(24, 8, RandomSpreadType.LINEAR, 20083232)
            )
        )
    }

    private fun register(pName: String): ResourceKey<StructureSet> {
        return ResourceKey.create(
            Registries.STRUCTURE_SET,
            Identifier.fromNamespaceAndPath(JAPANESE_MOD_ID, pName)
        )
    }
}
